pkg/controller/operators/catalog/operator.go

```diff
@@ -1731,6 +1732,28 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 
 				plan.Status.Plan[i].Status = status
 
+			case configMapKind:
```

Comment:
Won't this be caught by the `default` case now?

Reply:
the default case as it stands now calls `isSupported` which checks for supported custom resources only
```
// isSupported returns true if OLM supports this type of CustomResource.
func isSupported(kind string) bool {
	switch kind {
	case PrometheusRuleKind, ServiceMonitorKind:
		return true
	default:
		return false
	}
}
```
for configmaps this would return false and the install plan would fail. I think for core types having the explicit handling we have here is useful. If we use the dynamic client in the default case we lose the merge strategy control we have in the typed client

Reply:
I'm not sure this is too big a deal for the majority of cases, but it can be changed later should we decide to do so.

Reply:
so we should just change this PR to add ConfigMapKind to `isSupported` and call it a day?

Reply:
I still feel like there's value treating `ConfigMap` types similarly to the existing core types like secrets, service accounts, and roles in the code. leaving the default option for CR resources only

Reply:
I can't immediately see why we are treating those types differently today. Since it's the precedent,  I don't think there's harm in addressing this later.